packages/frontend/src/routes/blocks/blocks.route.js

```diff
@@ -22,50 +28,108 @@ function Blocks({blocks}) {
 export async function loader({params}) {
     const {blocksCount} = await Api.getStatus();
 
-    const blocks = await Api.getBlocks(blocksCount - blocksPerPage, blocksCount);
+    const blocks = await Api.getBlocks(blocksCount - blocksPerPageConfig.default, blocksCount);
 
     return {blocks, blocksCount};
 }
 
 function BlocksRoute() {
     const {blocks: defaultBlocks, blocksCount} = useLoaderData()
     const [blocks, setBlocks] = useState(defaultBlocks)
+    const [blocksPerPage, setBlocksPerPage] = useState(blocksPerPageConfig.default);
+    const [paginateActivePage, setPaginateActivePage] = useState(0);
+    const [searchedHeight, setSearchedHeight] = useState(0);
+    const [searchedHeightCorrection, setSearchedHeightCorrection] = useState(true);
 
     const pageCount = Math.ceil(blocksCount / blocksPerPage)
 
     const handlePageClick = async ({selected}) => {
-        const fromBlock = blocksCount - ((selected+1) * blocksPerPage)
+        const fromBlock = blocksCount - ((selected+1) * blocksPerPage) + 1;
         const toBlock = blocksCount - (((selected+1) - 1) * blocksPerPage)
         const updated = await Api.getBlocks(fromBlock, toBlock);
 
         setBlocks(updated)
     }
 
+    const goToHeightInputChangeHandle = (e) => {
+        const [lastBlock] = defaultBlocks;
```

Comment:
Don't use defaultBlocks as it is default value passed from the router's loader function